### PR TITLE
feat: Add Curvilinear distance

### PR DESCRIPTION
**Pull request by tepa46**

### CODE REVIEW

File: model/geom/simpleGeom.go [hunk 1]
```diff
@@ -15,16 +15,33 @@ func (geom *simpleGeom) MovePoint(point Point) {
...
-	point.set(x, y)
+	movedPoint := geom.LimitPointMovement(point, NewPoint(x, y))
+	movedPoint = geom.CorrectMovingAchievable(point, movedPoint, geom.newSimpleRandomPoint)
+
+	point.set(movedPoint.X(), movedPoint.Y())
+}
+
+func (geom *simpleGeom) newSimpleRandomPoint(_ Point) Point {
+	return geom.NewRandomPoint()
 }
```
Comment (Arsene-Baitenov) [inline model/geom/simpleGeom.go]: change sync rand to async

File: model/geom/curvilinearDistance.go [hunk 2]
```diff
@@ -1,6 +1,185 @@
 package geom
 
+import "math"
+
+// The distance between two points is considered to be half the length of the circle constructed through these points.
+// Cats want to see with both eyes, so achievability should be on both semicircles.
```
Comment (Arsene-Baitenov) [inline model/geom/curvilinearDistance.go:6]: lgtm

Comment (Arsene-Baitenov) [reply model/geom/curvilinearDistance.go:6]: lgbt

Comment (Arsene-Baitenov) [reply model/geom/curvilinearDistance.go:6]: abcdef

Comment (Arsene-Baitenov) [reply model/geom/curvilinearDistance.go:6]: lbtd

**Comment by Arsene-Baitenov** [comment]
Use special cf in geom to use rndAsync as "non-deterministic" function

**Comment by tepa46** [comment]
tormoza ne otkashut na spuske

**Comment by Arsene-Baitenov** [comment]
na podem ne zaglohnet motor

**Comment by tepa46** [comment]
I pomchit po uhabam po russkim

**Comment by Arsene-Baitenov** [comment]
dalnoboyshik, vodila, shofer

File: model/geom/curvilinearDistance.go (hunk 2 above)
Comment (Arsene-Baitenov) [reply model/geom/curvilinearDistance.go:6]: gun done

Comment (tepa46) [reply model/geom/curvilinearDistance.go:6]: thx

File: model/geom/simpleGeom.go (hunk 1 above)
Comment (tepa46) [reply model/geom/simpleGeom.go]: good idea

**Comment by Arsene-Baitenov** [comment]
cellByRowColumn?

**Review by Arsene-Baitenov — COMMENTED** [review]
looks so good, thx

**Comment by tepa46** [comment]
must be refactored in the this PR (by you!)

**Review by Arsene-Baitenov — APPROVED** [review]
.

**Pull request merged**